lib/segment/src/index/vector_index_base.rs

```diff
@@ -31,7 +31,17 @@ pub trait VectorIndex {
     ) -> OperationResult<Vec<Vec<ScoredPointOffset>>>;
 
     /// Force internal index rebuild.
-    fn build_index(&mut self, permit: Arc<CpuPermit>, stopped: &AtomicBool) -> OperationResult<()>;
+    fn build_index(&mut self, permit: Arc<CpuPermit>, stopped: &AtomicBool) -> OperationResult<()> {
+        self.build_index_with_progress(permit, stopped, || ())
+    }
+
+    /// Force internal index rebuild.
+    fn build_index_with_progress(
+        &mut self,
+        permit: Arc<CpuPermit>,
+        stopped: &AtomicBool,
+        tick_progress: impl FnMut(),
```

Comment:
Looks like it doesn't need to be `Mut`. If we use Atomics to count (and it looks like we do) - `impl Fn` should be enough

Reply:
I'd prefer to accept the broadest possible type of closure, which is `FnMut` in this case.

Reply:
It might be that having FnMut can prevent some compiler-level optimizations